src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -565,11 +565,25 @@ void CodegenNeuronCppVisitor::print_namespace_stop() {
     printer->pop_block();
 }
 
+void CodegenNeuronCppVisitor::append_conc_write_statements(
+    std::vector<ShadowUseStatement>& statements,
+    const Ion& ion,
+    const std::string& /* concentration */) {
+    auto ion_name = ion.name;
+    int dparam_index = get_int_variable_index(fmt::format("style_{}", ion_name));
 
-std::string CodegenNeuronCppVisitor::conc_write_statement(const std::string& ion_name,
-                                                          const std::string& concentration,
-                                                          int index) {
-    return "";
+    auto style_name = fmt::format("_style_{}", ion_name);
+    auto style_stmt = fmt::format("int {} = *(_ppvar[{}].get<int*>())", style_name, dparam_index);
+    statements.push_back(ShadowUseStatement{style_stmt, "", ""});
+
+
+    auto wrote_conc_stmt = fmt::format("nrn_wrote_conc(_{}_sym, {}, {}, {}, {})",
+                                       ion_name,
+                                       get_variable_name("ion_" + ion_name + "_erev"),
+                                       get_variable_name("ion_" + ion_name + "i"),
+                                       get_variable_name("ion_" + ion_name + "o"),
```

Comment:
A suggestion: for the `ion_name + "_erev"`, `ion_name + "i"`, etc parts: I thought we have methods in `struct Ion` to get these names. But I see that there is `ion.is_intra_cell_conc()` but no `ion.intra_cell_conc()`. Maybe we should add those in order to avoid these things getting hardcoded at multiple places?

I know there are already inconsistencies / hardcoded things. But just want to mention whenever I am seeing this aspect.